### PR TITLE
🔨 Maintenance/tools and non-critical fixes

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

A somehow *small* random collection of 🐛 non-critical fixes, ♻️ cleanup and 🔨 new tools added while developing other features but decided to move here to avoid shadowing other PRs.

- ✨ Minor improvements on email settings to avoid zero-length password (comes from my review on PR #2875)
- ⬆️ fixes deprecation warning after upgrade to latest ``aiohttp`` (from  PR #2867)
- 🔨 Adds dev script to run [pydeps](https://pydeps.readthedocs.io/en/latest/) and monitor module dependencies (from  PR #2867). Mostly to assist future refactoring projects plugin. Some examples
```cmd
scripts/pydeps.bash  --help
```
```cmd
scripts/pydeps.bash services/web/server/src/simcore_service_webserver \
     --only "simcore_service_webserver.projects" \
     --cluster
```
```cmd
scripts/pydeps.bash  services/web/server/src/simcore_service_webserver/ \
     --only "simcore_service_webserver.rest" \
     -T png \
     -o simcore_service_webserver.rest.ignore.png 
```

- 🔥 removes last references to ``APP_CONFIG_KEY`` (now only settings)

## Related issue/s

A split from large PR #2867 which will be closed w/o merge.

## How to test

- Covered by CI

### CODE REVIEW

File: scripts/pydeps.bash [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    --volume="$WORKDIR:/home/$USER/workdir" \
+    --user="$(id --user "$USER")":"$(id --group "$USER")" \
+    --entrypoint=pydeps \
+    "$IMAGE_NAME" \
+    "$@"
+
+}
+
+# Examples:
+#  - SEE  https://pydeps.readthedocs.io/en/latest/#usage
```
Comment (GitHK) [inline scripts/pydeps.bash:40]: very nice, thanks for putting this in

File: services/web/server/tests/unit/with_dbs/01/test_storage.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def storage_server(event_loop, aiohttp_server, app_cfg):
     app = create_safe_application(cfg)
 
     async def _get_locs(request: web.Request):
-        assert not request.has_body
+        assert not request.can_read_body
```
Comment (GitHK) [inline services/web/server/tests/unit/with_dbs/01/test_storage.py:23]: not sure I understand this change

**Review by GitHK — APPROVED** [review]

File: packages/settings-library/src/settings_library/email.py [hunk 3]
```diff
@@ -11,14 +11,15 @@
 class SMTPSettings(BaseCustomSettings):
     """Simple Mail Transfer Protocol"""
 
+    # TODO: NameEmail or EmailStr
```
Comment (sanderegg) [inline packages/settings-library/src/settings_library/email.py:14]: missing todo?

File: scripts/pydeps-docker/.pydeps [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[pydeps]
+max_bacon = 2
```
Comment (sanderegg) [inline scripts/pydeps-docker/.pydeps:2]: this is not vergan at all..

File: services/web/server/tests/unit/with_dbs/01/test_storage.py (hunk 2 above)
Comment (sanderegg) [reply services/web/server/tests/unit/with_dbs/01/test_storage.py:23]: that is in fact a deprecation of aiohttp...

**Review by sanderegg — APPROVED** [review]
Thanks for taking care of this!

File: services/web/server/tests/unit/with_dbs/01/test_storage.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/tests/unit/with_dbs/01/test_storage.py:23]: @GitHK , second point int he description above :-)

File: scripts/pydeps-docker/.pydeps (hunk 4 above)
Comment (pcrespov) [reply scripts/pydeps-docker/.pydeps:2]: That made me laugh :-D

File: packages/settings-library/src/settings_library/email.py (hunk 3 above)
Comment (pcrespov) [reply packages/settings-library/src/settings_library/email.py:14]: I left it as an exercise but did not dare to change even more...

**Pull request merged**